toothpick-kotlin/README.md

```diff
@@ -0,0 +1,115 @@
+# Toothpick Extensions  
+Toothpick Extensions
+
+### Base Extensions
+```kt
+// Inject from Scope
+scope.inject(obj)
+
+// Inject with Kotlin Class
+scope.getInstance(KotlinClass::class)
+// Inject with reified type
+val type: KotlinClass = scope.getInstance()
+
+// In a Module, bind with Kotlin Class
+module.bind(KotlinClass::class)...
+
+// Inject dependency without specifying type
+val dependency: Dependency = inject(scope)
+val dependency: Dependency = inject(scope = scope, name = "Name")
+val dependency: Dependency = inject(scope = scope, annotationName = AnnotationClass::class)
+
+// Lazy inject dependency via delegate
+val dependency: Dependency by injectLazy(scope)
+
+// Lazy inject with lazy scope assignment
+private lateinit var lateScope: Scope
+private lateinit var lateName: String
+
+private val dependency: Dependency by inject { scope = lateScope }
+private val namedDependency: Dependency by inject(name = "Name") { scope = lateScope }
+private val lateNamedDependency: Dependency by { scope = lateScope, name = lateName }
+
+// ...later
+scope = Toothpick.openScope(...)
+lateName = "myLateName"
+dependency.foo()           // Injection happens here
+namedDependency.foo()      // Injection happens here
+lateNamedDependency.foo()  // Injection happens here
+
+// Lazy inject with scope search
+class ScopeSearch : HasScope {
+    override lateinit var scope: Scope
+        protected set
```

Comment:
I would prefer an open get() that subclasses can override than a set.

Reply:
I'm not sure I follow the comment. I settled with this as it provided universal/public access to the getter but only protected access to the setter. This keeps any ole class changing the scope.